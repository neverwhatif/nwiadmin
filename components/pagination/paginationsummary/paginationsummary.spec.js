import React from 'react';
import { shallow } from 'enzyme';

import PaginationSummary from './index';

describe('PaginationSummary', () => {
    const component = shallow(<PaginationSummary from={1} to={10} total={50} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(component.text()).toBe('1 - 10 of 50');
    });
});
