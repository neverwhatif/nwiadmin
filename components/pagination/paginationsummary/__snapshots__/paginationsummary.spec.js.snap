// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationSummary should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaginationSummary
    from={1}
    to={10}
    total={50}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <strong>
          1
        </strong>,
        " - ",
        <strong>
          10
        </strong>,
        " of ",
        <strong>
          50
        </strong>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": "strong",
      },
      " - ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 10,
        },
        "ref": null,
        "rendered": 10,
        "type": "strong",
      },
      " of ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 50,
        },
        "ref": null,
        "rendered": 50,
        "type": "strong",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <strong>
            1
          </strong>,
          " - ",
          <strong>
            10
          </strong>,
          " of ",
          <strong>
            50
          </strong>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "strong",
        },
        " - ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 10,
          },
          "ref": null,
          "rendered": 10,
          "type": "strong",
        },
        " of ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 50,
          },
          "ref": null,
          "rendered": 50,
          "type": "strong",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
